### ISSUE TITLE
`subscribe()` does not chain properly

### ISSUE THREAD

**Issue by Dorus** [ReactiveX/rxjs#2283]
Subscribing to an observable with `subscribe()` instead of `subscribe(func)` makes the chain a bit shaky so to say.
```Js
Observable.of(null)
  .do(v => {
    console.log('a #1');
    Observable.of(null).subscribe(/* 1 */);
  })
  .do(v => {
    console.log('a #2'); // 2
  })
  .subscribe(/* 3 */);
```
The code at `// 2` does not run. It will output:
```text
a #1
```
but expected was:
```text
a #1
a #2
```

Notice that adding any function (for example `() => {}`) to `subscribe` at `/* 1 */` or `/* 3 */` will make it work as expected.

**Comment by Dorus** [comment]
I want to add that this pattern might look a bit weird, but it's a minimised testcase (thanks to @mjamin ) and multiple Angular 2 users have ran into this problem lately.

**Comment by mjamin** [comment]
Here's the plunker: https://plnkr.co/edit/E7SIfy4h2JRt5sytDLFw.

It's a heavily minified testcase. E.g. the observable inside the first `do` is a http post request that doesn't care about the response body.

**Comment by trxcllnt** [comment]
Duplicate of #2237. Will try to have a fix in today.

**Comment by Dorus** [comment]
Has this been fixed yet? Yet another user in the Gitter chat reported very similar problems using `5.1.x` today.

**Comment by maxime1992** [comment]
@Dorus I think you're talking about me on Gitter and I opened an other [issue](https://github.com/Reactive-Extensions/RxJS/issues/1423) (with repro) but probably a dup of this one.

**Comment by kwonoj** [comment]
@Dorus check https://github.com/ReactiveX/rxjs/issues/2237#issuecomment-281228121 - PR is in review and latest published package doesn't include it yet.

**Comment by kwonoj** [comment]
closed via #2238.
